File: app-presentation/src/main/java/com/zxjdev/smile/presentation/moment/list/adapter/MomentAdapter.java
package com.zxjdev.smile.presentation.moment.list.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.zxjdev.smile.R;
import com.zxjdev.smile.presentation.common.util.image.ImageLoader;
import com.zxjdev.smile.presentation.moment.MomentModel;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class MomentAdapter extends RecyclerView.Adapter<MomentViewHolder> {

  private List<MomentModel> moments = new ArrayList<>();
  private ImageLoader imageLoader;

  public MomentAdapter(ImageLoader imageLoader) {
    this.imageLoader = imageLoader;
  }

  @Override
  public MomentViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    LayoutInflater inflater = LayoutInflater.from(parent.getContext());
    View itemView = inflater.inflate(R.layout.item_moment, parent, false);
    return new MomentViewHolder(itemView);
  }

  @Override
  public void onBindViewHolder(MomentViewHolder holder, int position) {
    MomentModel moment = moments.get(position);
    holder.tvContent.setText(moment.getContent());
    holder.tvName.setText(moment.getOwner().getUsername());
    holder.tvTime.setText(parseCreateDate(moment.getCreateAt()));

    imageLoader.loadCircleImage(moment.getOwner().getAvatar(), holder.ivAvatar);
  }

  @Override
  public int getItemCount() {
    return moments.size();
  }

  public void setMoments(List<MomentModel> moments) {
    this.moments = moments;

    notifyDataSetChanged();
  }

  private String parseCreateDate(Date date) {
    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.CHINA);
    return dateFormat.format(date);
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/moment/list/adapter/MomentAdapter.kt
package com.zxjdev.smile.presentation.moment.list.adapter

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.zxjdev.smile.R
import com.zxjdev.smile.presentation.common.util.image.ImageLoader
import com.zxjdev.smile.presentation.moment.MomentModel
import kotlinx.android.synthetic.main.item_moment.view.*
import java.text.SimpleDateFormat
import java.util.*

class MomentAdapter(private val imageLoader: ImageLoader) : RecyclerView.Adapter<MomentViewHolder>() {

    private var moments: List<MomentModel> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MomentViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val itemView = inflater.inflate(R.layout.item_moment, parent, false)
        return MomentViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MomentViewHolder, position: Int) {
        val moment = moments[position]
        holder.itemView.tv_content.text = moment.content
        holder.itemView.tv_name.text = moment.owner!!.username
        holder.itemView.tv_time.text = parseCreateDate(moment.createAt)

        imageLoader.loadCircleImage(moment.owner!!.avatar, holder.itemView.iv_avatar)
    }

    override fun getItemCount(): Int {
        return moments.size
    }

    fun setMoments(moments: List<MomentModel>) {
        this.moments = moments

        notifyDataSetChanged()
    }

    private fun parseCreateDate(date: Date?): String {
        val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.CHINA)
        return dateFormat.format(date)
    }
}

